### PR TITLE
Use detect-secrets to scan secrets (WIP)

### PR DESCRIPTION
**Pull request by liyongsea**

### CODE REVIEW

File: data_analysis/pii/detect_secret.ipynb [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      "135\n",
+      "docs/plans/big-data/testing.py\n",
+      "[{'type': 'Hex High Entropy String', 'line_number': 237, 'secret_value': '2988c44780790e4ffceb1f97391e475f165e316f27a656957282a2998aee9d4f'}, {'type': 'Hex High Entropy String', 'line_number': 236, 'secret_value': 'fa4e6aa7e7edaa6feb036fd5e8c28ffc48575cefc332187552c5be4bf0511af8'}, {'type': 'Hex High Entropy String', 'line_number': 245, 'secret_value': '46dabc02b59be44064a9e06dd50bc6841833578c2b6339fbc43f090cc17831fa'}, {'type': 'Hex High Entropy String', 'line_number': 248, 'secret_value': '9b4a551a6c1c5830d6070b9c22ae1788b9743e9637be47d56103bcda019a897c'}, {'type': 'Hex High Entropy String', 'line_number': 229, 'secret_value': '374c02504f89ed0a760b03c3e1fd2258988576b919d763254709b66fc7bfb22e'}, {'type': 'Hex High Entropy String', 'line_number': 228, 'secret_value': 'd07050610c50de8c810aa1d1e322786ed8932cf6eafa0fbe1f132b2c881af9c2'}, {'type': 'Hex High Entropy String', 'line_number': 249, 'secret_value': '9820f1ab795db7b0d195f21966ecb071d76e9ce2fd3a90845974a3905584eb3e'}, {'type': 'Hex High Entropy String', 'line_number': 244, 'secret_value': '983730afb7ab41d524b72f1097daaf4a3c15b98943291f96e523730849cabe8c'}]\n",
```
Comment (ocramz) [inline data_analysis/pii/detect_secret.ipynb:149]: Seems like there are a few false positives in these results. Is this intentional?

File: data_analysis/pii/analysis.ipynb [hunk 2]
```diff
@@ -562,6 +568,398 @@
...
+      "===============195==================\n",
+      "('372f0f95-6b48-487a-a1ac-383ba580fc1c', (1951, 1988), 'regex.Regex(\\'((?:(?:[A-Za-z]+[\\\\\\\\p{Nd}\\\\\\\\p{Pd}\\\\\\\\/\\\\\\\\+\\\\\\\\=:_]+|[\\\\\\\\p{Nd}\\\\\\\\p{Pd}\\\\\\\\/\\\\\\\\+\\\\\\\\=:]+[A-Za-z]+)){10,})(?:$|[\\\\\\\\b\\\\\\\\s\\\\\\\\p{Han}@?,!;:\\\\\\\\\\\\\\'\\\\\\\\\")(.])\\', flags=regex.M | regex.V0)', 'KEY')\n",
+      "\n",
+      "===============195==================\n",
+      "('372f0f95-6b48-487a-a1ac-383ba580fc1c', (3837, 3874), 'regex.Regex(\\'((?:(?:[A-Za-z]+[\\\\\\\\p{Nd}\\\\\\\\p{Pd}\\\\\\\\/\\\\\\\\+\\\\\\\\=:_]+|[\\\\\\\\p{Nd}\\\\\\\\p{Pd}\\\\\\\\/\\\\\\\\+\\\\\\\\=:]+[A-Za-z]+)){10,})(?:$|[\\\\\\\\b\\\\\\\\s\\\\\\\\p{Han}@?,!;:\\\\\\\\\\\\\\'\\\\\\\\\")(.])\\', flags=regex.M | regex.V0)', 'KEY')\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
```
Comment (ocramz) [inline data_analysis/pii/analysis.ipynb:953]: could you add a title or a comment re. what this snippet is doing?

**Review by ocramz — CHANGES_REQUESTED** [review]
It's a bit terse :) Personally I would appreciate more comments and/or section headers in the notebook.

**Comment by loubnabnl** [comment]
Yes I think this is a WIP as we'll experiment with this tool using other configs, see this [issue](https://github.com/bigcode-project/bigcode-dataset/issues/17), @liyongsea  maybe you can add WIP to the PR title?

File: data_analysis/pii/analysis.ipynb (hunk 2 above)
Comment (liyongsea) [reply data_analysis/pii/analysis.ipynb:953]: Sorry for that. It is still work in progress. I share the PR so if others want to run some experiment they can do it. I will still do some modifications and improve the code quality

**Comment by liyongsea** [comment]
Sorry for the confusion. It is WIP, so that people can run some tests if they want

File: data_analysis/pii/evaluate.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            pred_matched[pred_idx] = True
+            ref_idx += 1
+            pred_idx += 1
+        elif ref_interval[0] < pred_interval[0]:
+            ref_idx += 1
+        else:
+            pred_idx += 1
+    metrics = {
+        'TP_ref': sum(ref_matched),
+        'TP_pred': sum(pred_matched),
```
[the remark this answers is not in the document]
Comment (liyongsea) [reply data_analysis/pii/evaluate.py:37]: Nice catch !!!

it is not the same in some cases, for example:
- ref = [(0, 1), (2, 3)]
- pred = [(1, 10)]
Here TP_ref = 2, TP_pred = 1
However, our friend **copilot** make a mistake here, `elif ref_interval[0] < pred_interval[0]:` shoud be `if ref_interval[1] < pred_interval[1]:`

**Pull request closed without merging**